src/Memcache.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace TraderInteractive\Memoize;
+
+/**
+ * A memoizer that caches the results in memcache.
+ */
+class Memcache implements Memoize
+{
+    /**
+     * The memcache client
+     *
+     * @var \Memcache
+     */
+    private $client;
+
+    /**
+     * Cache refresh
+     *
+     * @var boolean
+     */
+    private $refresh;
+
+    /**
+     * Sets the memcache client.
+     *
+     * @param \Memcache $client  The memcache client to use
+     * @param boolean         $refresh If true we will always overwrite cache even if it is already set
+     */
+    public function __construct(\Memcache $client, bool $refresh = false)
+    {
+        $this->client = $client;
+        $this->refresh = $refresh;
+    }
+
+    /**
+     * The value is stored in memcache as a json_encoded string,
+     * so make sure that the value you return from $compute is json-encode-able.
+     *
+     * @see Memoize::memoizeCallable
+     *
+     * @param string   $key
+     * @param callable $compute
+     * @param int|null $cacheTime
+     *
+     * @return mixed
+     */
+    public function memoizeCallable(string $key, callable $compute, int $cacheTime = null)
+    {
+        if (!$this->refresh) {
+            try {
```

Comment:
Update to support newer method signature
```suggestion
     * @param int|null $cacheTime
     * @param bool     $refresh
     *
     * @return mixed
     */
    public function memoizeCallable(string $key, callable $compute, int $cacheTime = null, bool $refresh = false)
    {
        if (!$this->refresh && !$refresh) {
            try {
```